File: ts/面试01-01.ts
function isUnique(astr: string): boolean {
    let mask = 0
    for (let c of astr) {
        let move = c.charCodeAt(0) - 'a'.charCodeAt(0)
        if ((mask & (1 << move)) !== 0) {
            return false
        }
        mask |= (1 << move)
    }
    return true
};

File: rs/面试01-01.rs
impl Solution {
    pub fn is_unique(astr: String) -> bool {
        let mut mask = 0;
        for c in astr.bytes() {
            let bit = 1 << (c - b'a');
            if mask & bit != 0 {
                return false;
            }
            mask |= bit;
        }
        true
    }
}